### PR TITLE
Szworkshop

### PR DESCRIPTION
**Pull request by cvicentiu**
This is our brilliant plugin code!

Contribution is contributed under BSD-New License. (Otherwise @svoj  will ask us for it).

### CODE REVIEW

**Comment by cvicentiu** [comment]
Please merge this! :)

File: plugin/proc_info/proc_info.cc [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+	DIR *dir; /* from dirent.h */
+	struct dirent dirbuf, *result;
+	TABLE *table= tables->table;
+
+	char final_result[FN_REFLEN + 1];
+	int error= 0;
+
+	if ((fd= open("/proc/self/fd", O_DIRECTORY)) < 0)
+		return 1;
```
Comment (svoj) [inline plugin/proc_info/proc_info.cc]: Please fix coding style!!!

Comment (cvicentiu) [reply plugin/proc_info/proc_info.cc]: Okay :(

File: plugin/proc_info/proc_info.cc [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  { 0, 0, MYSQL_TYPE_STRING, 0, 0, 0, 0 }
+};
+
+
+static int meminfo_fill(MYSQL_THD thd, TABLE_LIST *tables, COND *cond)
+{
+  TABLE *table= tables->table;
+  char name[1024];
+  unsigned long value;
+  FILE *fp;
```
Comment (vaintroub) [inline plugin/proc_info/proc_info.cc]: MYSQL_TYPE_LONG is int32. Probably not sufficient to describe virtual memory, even in kilobytes.

File: plugin/proc_info/proc_info.cc [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+#include <mysql/plugin.h>
+
+/* We'll use this to read directories */
+#include <dirent.h>
+
+
+static ST_FIELD_INFO meminfo_fields[]=
+{
+  { "NAME", 100, MYSQL_TYPE_STRING, 0, 0, 0, SKIP_OPEN_TABLE },
+  { "VALUE", 21, MYSQL_TYPE_LONG, 0, MY_I_S_UNSIGNED, 0, SKIP_OPEN_TABLE },
```
Comment (vaintroub) [inline plugin/proc_info/proc_info.cc]: MYSQL_TYPE_LONG is int32. Probably not sufficient to describe virtual memory, even in kilobytes. I think MYYSQL_TYPE_LONGLONG is more appropriate. also  please consider a longlong C datatype when parsing meminfo later when filling table.

**Comment by svoj** [comment]
Example PR, not to be merged.

**Pull request closed without merging**